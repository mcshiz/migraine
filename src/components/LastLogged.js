import React from 'react';
import 'moment'
import moment from "moment/moment";

class LastLogged extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
          lastLogged: null
        };
        this.getLastLogged = this.getLastLogged.bind(this)
    }

    getLastLogged() {
        let data = this.props.data;
        if(data.length <= 0 ) {
            return;
        }
        let lastLogged = new Date(Math.max.apply(null, data.map(function(e) {
            return new Date(e.date);
        })));
        if(Date.parse(lastLogged) && moment(lastLogged).isValid()) {
            return moment(lastLogged).format('MM/DD/YY h:mm:ss a');
        }
    }
    render () {
        const LastLogged = this.getLastLogged();

        return (
            <b className='last-logged'>Last Logged: {LastLogged}</b>
        )
    }
}
export default LastLogged