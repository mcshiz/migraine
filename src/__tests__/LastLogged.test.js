import React from 'react'
import LastLogged from '../components/LastLogged'

describe('LastLogged Component', () => {
    loadExampleData();
    const wrapper = shallow(<LastLogged data={exampleData}/>);
    it('Should load example data', () => {
       expect(wrapper.find('.last-logged').length).toBe(1);
    });
    it('Should compare the component with a snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

});