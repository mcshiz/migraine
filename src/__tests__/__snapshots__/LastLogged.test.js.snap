// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastLogged Component Should compare the component with a snapshot 1`] = `
<b
  className="last-logged"
>
  Last Logged: 
  12/29/17 9:33:33 am
</b>
`;
